File: summer2019/week2/design-hashmap.java
class MyHashMap {
    int[] keys;
    int[] vals;
    
    /** Initialize your data structure here. */
    public MyHashMap() {
        keys = new int[1000000];
        vals = new int[1000000];
    }
    
    /** value will always be non-negative. */
    public void put(int key, int value) {
        keys[key] = 1;
        vals[key] = value;
    }
    
    /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
    public int get(int key) {
        if (keys[key] <= 0) {
            return -1;
        }
        return vals[key];
    }
    
    /** Removes the mapping of the specified value key if this map contains a mapping for the key */
    public void remove(int key) {
        keys[key] = -1;
        vals[key] = -1;
    }
}

File: 2026/week2/design-hashmap.py
class MyHashMap:

    def __init__(self):
        self.vals = [[] for _ in range(1001)]

    def put(self, key: int, value: int) -> None:
        b = key // 1000
        i = key % 1000
        if len(self.vals[b]) == 0:
            self.vals[b] = [-1 for _ in range(1000)]
        self.vals[b][i] = value

    def get(self, key: int) -> int:
        b = key // 1000
        i = key % 1000
        if len(self.vals[b]) == 0:
            return -1
        return self.vals[b][i]

    def remove(self, key: int) -> None:
        b = key // 1000
        i = key % 1000
        if len(self.vals[b]) != 0:
            self.vals[b][i] = -1

# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)
